blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/astyanax/AstyanaxWriter.java

```diff
@@ -305,10 +364,22 @@ public static boolean isLocatorCurrent(Locator loc) {
         return insertedLocators.getIfPresent(loc.toString()) != null;
     }
 
+    public static boolean isDelayedLocatorForASlotCurrent(int slot, Locator locator) {
```

Comment:
I think you are correct with `DiscoveryWriter`--it is a bug.  I think it will probably be easier to override `ABasicMetricsRW.isLocatorCurrent()` and `APreaggregatedMetricsRW.isLocatorCurrent()` to use the cache in `AstyanaxWriter`.  

When we finally move to DataStax, we can delete all the Astyanax stuff.